mm2src/mm2_bitcoin/keys/src/segwitaddress.rs

```diff
@@ -150,10 +153,13 @@ impl FromStr for SegwitAddress {
         };
         // decode as bech32, should use Variant if Bech32m is used alongside Bech32
         // The improved Bech32m variant described in [BIP-0350](https://github.com/bitcoin/bips/blob/master/bip-0350.mediawiki)
-        let (_, payload, _) = bech32::decode(s)?;
+        let (_, payload, variant) = bech32::decode(s)?;
         if payload.is_empty() {
             return Err(Error::EmptyBech32Payload);
         }
+        if variant == bech32::Variant::Bech32m {
```

Comment:
If the new variant is added, it will pass this check and the address will be parsed incorrectly. Please add an explicit match statement here *without* `_` placeholder. We should get a compiler error when a new variant appears.